System/DataModule.cpp

```diff
@@ -1,764 +1,497 @@
+				// No, hit a collision and aren't allowed to overwrite it; entity wasn't added
+			} else {
+				return false;
+			}
+		} else {
+			// Ok, no exact collision, so now check all the parent classes for collisions. If there are none, add the entity to them all.
+
+			/* Don't do this; multiple instances of the same name are allowed on the same type typelist, as long as there isn't more than one whose EXACT type is that of the typelist's type name
+
+			// Walk up the class hierarchy till we reach the top, checking each level for whether there is an entity of the same or more general type, and same name
+			for (const Entity::ClassInfo *pClass = &(pEntToAdd->GetClass()); pClass != 0; pClass = pClass->GetParent()) {
+				// If any collisions, then report that we couldn't add the entity and quit
+				if (GetEntityIfExactType(pClass->GetName(), pEntToAdd->GetPresetName())) { return false; }
+			}
+			// Ok, doesn't seem to be any collisions, so just add the new entity */
+
+			// Let the entity know which module it belongs to
+			pEntToAdd->SetModuleID(m_ModuleID);
+			// Make the copy which will become the preset instance
+			Entity *pEntClone = pEntToAdd->Clone();
+			// Mark the one we are about to add to the list as the Original now - this is now the actual Original Preset instance
+			pEntClone->m_IsOriginalPreset = true;
+
+			// Insert clone into the list of entries where it will now be owned
+			// New file location specified
+			if (readFromFile != "Same") {
+				m_PresetList.push_back(PresetEntry(pEntClone, readFromFile));
+				m_EntityList.push_back(pEntClone);
+
+				// If same file specified for new instance, use the one in the last entry
+			} else if (!m_PresetList.empty()) {
+				m_PresetList.push_back(PresetEntry(pEntClone, m_PresetList.back().m_FileReadFrom));
+				m_EntityList.push_back(pEntClone);
+
+			} else {
+				RTEAbort("Tried to add first entity instance to data module " + m_FileName + " without specifying a data file!");
+			}
+			// Add it to the type map and quit
+			added = AddToTypeMap(pEntClone);
+			RTEAssert(added, "Unexpected problem while adding Entity instance \"" + pEntToAdd->GetPresetName() + "\" to the type map of data module: " + m_FileName);
+		}
+		return added;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool DataModule::GetGroupsWithType(std::list<std::string> &groupList, std::string withType) {
+		bool foundAny = false;
+
+		if (withType == "All" || withType.empty()) {
+			// Just give all the groups
+			for (std::list<std::string>::const_iterator gItr = m_GroupRegister.begin(); gItr != m_GroupRegister.end(); ++gItr) { groupList.push_back(*gItr); }
+		} else {
+			// Go through all entities, adding the groups the ones of the right type belong to
+			// Find the type entry in the type map
+			map<std::string, std::list<std::pair<std::string, Entity *> > >::iterator clsItr = m_TypeMap.find(withType);
```

Comment:
The comments above are junk.
Also oneline early return, spacing before `>`s, bad iterator name